app_server/app/controllers/ViewController.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * View Controller
+ * @module ViewController
+ */
+var rfr = require('rfr');
+var Joi = require('joi');
+var Boom = require('boom');
+var Promise = require('bluebird');
+
+var CustomError = rfr('app/util/Error');
+var Utility = rfr('app/util/Utility');
+var Service = rfr('app/services/Service');
+var Authenticator = rfr('app/policies/Authenticator');
+
+var logger = Utility.createLogger(__filename);
+
+function ViewController(server, options) {
+  this.server = server;
+  this.options = options;
+}
+
+var Class = ViewController.prototype;
+
+Class.registerRoutes = function() {
+
+  this.server.route({method: 'GET', path: '/{id}',
+                     config: {
+                       validate: singleViewValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.getListOfUsersViewingStream});
+
+  this.server.route({method: 'POST', path: '/{id}',
+                     config: {
+                       validate: singleViewValidator,
+                       auth: {scope: Authenticator.SCOPE.ALL}
+                     },
+                     handler: this.createView});
+};
+
+/* Routes handlers */
+Class.createView = function(request, reply) {
+  logger.debug('Creating View');
+
+  var userId = request.auth.credentials.userId;
+  var streamId = request.params.id;
+
+  Service.createView(userId, streamId).then(function receiveResult(result) {
+    if (result instanceof CustomError.NotFoundError) {
+      logger.error('View could not be created');
+      reply(Boom.badRequest(result.message));
+      return;
+    }
```

Comment:
Would be nicer to do this in 1 statement: `return reply(Boom.badRequest(result.message));`